File: 01/solve.py
with open('./input', 'r') as f:
    numbers = []
    for l in f:
        numbers.append(int(l.strip()))
    
def part1(nums):
    count = 0
    for i in range(1, len(nums)):
        if nums[i] > nums[i-1]:
            count += 1
    return count

print(part1(numbers))

def part2(nums):
    count = 0
    for i in range(3, len(nums)):
        if sum(nums[i-2:i+1]) > sum(nums[i-3:i]):
            count += 1
    return count

print(part2(numbers))

File: 01/solve.js
const fs = require('fs')

fs.readFile('./input','utf-8', (err, data) => {
    let nums = data.split('\n').map(Number);
    
    let solution = nums.reduce((count, current, index, arr) => {
        if(arr[index+1] > current){
            return count +=1;
        } else {
            return count;
        }
    }, 0);
    
    console.log(solution);

})

fs.readFile('./input','utf-8', (err, data) => {
    let nums = data.split('\n').map(Number);
    
    let solution = nums.reduce((count, current, index, arr) => {
        if(arr[index+3] > current){
            return count +=1;
        } else {
            return count;
        }
    }, 0);
    
    console.log(solution);

})